### PR TITLE
Change `query_registration()` to use `_get_v2_account()`

### LINKED ISSUE #9306: `show_account` broken in combination with ACMEv1 account reuse using symbolic links
Reported on the Community here: https://community.letsencrypt.org/t/certbot-show-account-fails-with-acme-v01-api-letsencrypt-org-dns-lookup-failure/178367

Since #6156 ACMEv1 account reuse is done using symbolic links. 

In #9127 the `show_account` subcommand was introduced, making use of the [`query_registration`](https://github.com/certbot/certbot/blob/5c111d0bd1206d864d7cb93754e101f6073bc669/acme/acme/client.py#L641-L655) function of the `acme` library, passing the `regr` object from the account to that function.

However, when symbolic link account reuse is in place, the `regr` object contains an account `uri` of the ACMEv1 protocol, including the now-nonexisting `acme-v01` API URL. That URI is directly used in [`_send_recv_regr()`](https://github.com/certbot/certbot/blob/5c111d0bd1206d864d7cb93754e101f6073bc669/acme/acme/client.py#L84), triggering a DNS error.

Possible fixes:

* Detect ACMEv1 account URI and translate it into an ACMEv2 account URI *in Certbots `main.show_account`* and keep using `query_registration`, or
* Detect ACMEv1 account URI and translate it into an ACMEv2 account URI *in acme's `client.query_registration`*, or
* Rewrite Certbots `main.show_account()` to *not* use `query_registration` but build a signed request for `newAccount` on its own.

### PR DESCRIPTION
**Pull request by osirisinferi**
Fixes #9306.

I also needed to update `_get_v2_account()` to include the registrations resources body.

It seems the `terms_of_service=regr.terms_of_service` part in the original code isn't necessary any longer. With these changes, when I set `terms_of_service` to a random value in `query_registration()` before the call to `_get_v2_account()` it still is that random value *after* the call.. So I guess it's not required. Not sure if the unittests also check this.

Another thing: maybe `regr = self._regr_from_response(response)` makes more sense in `_get_v2_account()` too, instead of the `regr.update()` currently used?

### CODE REVIEW

**Review by alexzorin — COMMENTED** [review]
This breaks `certbot update_account -m email@example.com`, because the signature of `update_registration` is:

```python
    def update_registration(self, regr: messages.RegistrationResource,
                            update: Optional[messages.Registration] = None
                            ) -> messages.RegistrationResource:
```

The first argument being the existing `regr`, and  the second argument being the changes you actually want to make.

Unfortunately, `main.update_account` calls `update_registration(regr.update(...))`, instead of `update_registration(regr, regr.body.update(contact=...))`.

The former method is broken by this PR, because `_get_v2_account` overwrites the body of the `regr` (the first argument) with what is sent by the ACME server. This wipes out the change to `contact` queued up by `main.update_account`. 

This is correct behavior in my opinion, but arguably a breaking change to the `acme` module's API.

We could fix the way `main.update_account` calls `update_registration`, but we'd still be potentially breaking other API users who were relying on the old behavior of `update_registration`.

@ohemorange do you have any thoughts? I think you might have most recently been dealing with this part of the code.

**Comment by osirisinferi** [comment]
Not sure, but wouldn't a check if `body` already exists before overwriting it in line 675 in this PR be a (temporary) solution? The only reason I needed to do this, is because `regr.body` was missing. If it isn't missing, I don't need to add/overwrite it.

**Comment by alexzorin** [comment]
Yes, perhaps we can retain the previous behavior if no `update` is provided. 

I am perplexed as to what `update_registration` is meant to do if no `update` is passed, but by the looks of https://github.com/certbot/certbot/pull/706 I think it's just a regrettable part of the original API design. 

Maybe go ahead and give that a try, I'd still be curious to hear what Erica thinks though.

**Comment by osirisinferi** [comment]
I actually don't think the changes in e88a23a are a good idea. The whole idea for this PR is to improve compatibility with older ACMEv1 accounts which are used for the ACMEv2 API due to symlinking. e88a23a works for ACMEv2 accounts due to the fact those accounts have an empty `regr.json` locally. However, ACMEv1 accounts have a non-empty `regr.json` filled with local account info. Consequently e88a23a doesn't work for those ACMEv1 accounts: up to date info from the ACME server wouldn't overwrite the local info from `regr.json`.

I'm now thinking more to add an extra parameter to `_get_v2_account()`: `update_body = False` for example. `query_registration()` would set this argument to `True` and make the logic in `_get_v2_account()` to simply use `update_body` instead of the whole "is `regr.body` empty or not" logic.

File: certbot/CHANGELOG.md [hunk 1]
```diff
@@ -16,7 +16,9 @@ Certbot adheres to [Semantic Versioning](https://semver.org/).
 
 ### Fixed
 
-*
+* The `show_account` now uses the "newAccount" ACME endpoint to fetch the account data, so it
```
Comment (alexzorin) [inline certbot/CHANGELOG.md:19]: New Certbot has been released, please merge `master` and move this entry to the new section.

**Review by alexzorin — CHANGES_REQUESTED** [review]
I like it! 👍 from me.

I wish I had some ACMEv1/ACMEv2 data to test this with.

**Review by alexzorin — APPROVED** [review]
LGTM. I like the simplicity, in the end. Thanks for fixing this.

**Pull request merged**